Linear.lua

```diff
@@ -22,25 +23,28 @@ function Linear:reset(stdv)
          self.weight:select(1, i):apply(function()
             return torch.uniform(-stdv, stdv)
          end)
-         self.bias[i] = torch.uniform(-stdv, stdv)
+      end
+      if self.bias then
+         for i=1,self.bias:nElement() do
+            self.bias[i] = torch.uniform(-stdv, stdv)
+         end
       end
    else
       self.weight:uniform(-stdv, stdv)
-      self.bias:uniform(-stdv, stdv)
+      if self.bias then self.bias:uniform(-stdv, stdv) end
    end
-
    return self
 end
 
 function Linear:updateOutput(input)
    if input:dim() == 1 then
-      self.output:resize(self.bias:size(1))
-      self.output:copy(self.bias)
+      self.output:resize(self.weight:size(1))
+      if self.bias then self.output:copy(self.bias) else self.output:fill(0) end
```

Comment:
Just wondering, isn't `zero()` faster than `fill(0)` for contiguous tensors in GPU ?